sdv/constraints/tabular.py

```diff
@@ -334,6 +334,14 @@ def _transform(self, table_data):
             pandas.DataFrame:
                 Transformed data.
         """
+        # To make the NaN to None mapping work for pd.Categorical data, we need to convert
+        # the columns to object before replacing NaNs with None.
+        table_data[self._columns] = table_data[self._columns].astype({
+            col: object
+            for col in self._columns
+            if pd.api.types.is_categorical_dtype(table_data[col])
+        })
```

Comment:
Based on [this discussion](https://github.com/pandas-dev/pandas/issues/40472), should we just use fillna instead of replace? Then we don't have to convert

Reply:
@R-Palazzo I forgot that fillna cannot be used with None which is probably why we have this line of code in the first place. This solution is good